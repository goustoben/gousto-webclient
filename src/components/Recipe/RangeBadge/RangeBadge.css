@value White, Pomegranate from 'styles/colors.css';

.rangeBadge{
  position: relative;
  height:28px;
  display: flex;
  flex-direction: row;
}


.ribbon {
  width: 10px;
  position: relative;
  display: inline-flex;
  flex-direction: column;
  border-top: 1px solid White;
  border-bottom: 1px solid White;
}

.arrow {
  height: 28px;
  position: absolute;
  z-index: 2;
	width: 0;
	height: 0;
}

.arrowTopBorder  {
	composes: arrow;
  border-right: none;
  border-top: none;
  border-bottom: 16px solid transparent;
  border-left: 10px solid White;
}

.arrowTop {
  composes: arrow;
  border-right: none;
  border-top: none;
  border-bottom: 14px solid transparent;
  border-left: 9px solid Pomegranate;
}

.arrowTop::before,
.arrowBottom::before {
  content: '';
  display: none;
  width: 100%;
  height: 100%;
  position: absolute;
  right: 0;
  border-right: none;
  border-left: 9px solid rgba(0,0,0,0.1);
}

.arrowTop::before {
  top: 0;
  border-top: none;
  border-bottom: 14px solid transparent;
}

.arrowBottom::before {
  bottom: 0;
  border-top: 14px solid transparent;
  border-bottom: none;
}

.ribbonTextHighlight:hover + .ribbon .arrowTop::before,
.ribbonTextHighlight:hover + .ribbon .arrowBottom::before,
.ribbonTextHighlight:focus + .ribbon .arrowTop::before,
.ribbonTextHighlight:focus + .ribbon .arrowBottom::before {
  display: block;
}

.arrowBottom {
	composes: arrow;
  top: 12px;
  border-right: none;
  border-top: 14px solid transparent;
  border-bottom: none;
  border-left: 9px solid Pomegranate;
}

.arrowBottomBorder {
	composes: arrowBottom;
  right: -1px;
  border-top: 15px solid transparent;
  border-bottom: none;
  border-left: 10px solid White;
}

.ribbonText {
  z-index: 2;
  display: inline-flex;
  align-items: center;
  height: 100%;
  justify-content: center;
  background-color: Pomegranate;
  border: 1px solid White;
  border-right: none;
  color: White;
  padding: 5px 11px 5px 15px;
  font-size: 12px;
  font-weight: 700;
  font-family: Avenir-Black, Avenir;
  position: relative;
  &:focus {
    outline: 0;
  }
}

.ribbonTextHighlight::before {
    content: '';
    display: none;
    height: 100%;
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    background: rgba(0, 0, 0, 0.1);
    z-index: 1;
}

.ribbonTextHighlight:hover::before,
.ribbonTextHighlight:focus::before {
  display: block;
  outline: 0;
}

.ribbonText:hover .foodBrandNameUnderlined,
.ribbonText:focus .foodBrandNameUnderlined
 {
    text-decoration: underline;
    outline: 0;
}

.foodBrandName {
  position: relative;
  z-index: 2;
  text-transform: uppercase;
}

.rightChevron {
  composes: fa-basic-class fa-fw fa-chevron-right from 'styles/vendor/font-awesome-module.css';
  font-size: 11px;
  z-index: 2;
  height: 100%;
  padding-top: 3px;
  padding-left: 6px;
}
