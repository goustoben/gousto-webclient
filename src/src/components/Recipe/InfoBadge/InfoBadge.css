@value Haricot, Radish, Carrot, White, DarkChilli, White from 'styles/colors.css';

.badge {
	composes: paddingSMLeftRight marginXSRight marginXSBottom from 'sass-loader!styles/spacing.scss';
	composes: inlineBlock from 'styles/display.css';
	composes: textXS from 'styles/typography.css';
	font-weight: bold;
	border-radius: 1.4em;
	border: 1px solid
}

.newRecipe {
	color: Haricot;
	border-color: Haricot;
}

.stockInfo {
	color: Radish;
	border-color: Radish;
}

.recommended {
	color: Carrot;
	border-color: Carrot;
}

.inverse {
	color: White;
	border: none;
	background-color: DarkChilli;
}
