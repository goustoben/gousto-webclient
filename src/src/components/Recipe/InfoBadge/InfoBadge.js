import React from 'react'
import classNames from 'classnames'
import css from './InfoBadge.css'

const InfoBadge = ({ type = 'span', children, newRecipe, stockInfo, recommended, inverse, ...props }) => {
	const className = classNames(
		css.badge,
		{ [css.newRecipe]: newRecipe },
		{ [css.stockInfo]: stockInfo },
		{ [css.recommended]: recommended },
		{ [css.inverse]: inverse },
	)

	return React.createElement(type, { className, ...props }, children)
}

InfoBadge.propTypes = {
	type: React.PropTypes.string,
	children: React.PropTypes.node,
	newRecipe: React.PropTypes.bool,
	stockInfo: React.PropTypes.bool,
	recommended: React.PropTypes.bool,
	inverse: React.PropTypes.bool,
}

InfoBadge.defaultProps = {
	newRecipe: false,
	stockInfo: false,
	recommended: false,
	inverse: false,
}

export default InfoBadge
