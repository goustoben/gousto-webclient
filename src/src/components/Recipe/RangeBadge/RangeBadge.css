@value White, Pomegranate from 'styles/colors.css';

.rangeBadge{
  position: relative;
  height:28px;
  display: flex;
  flex-direction: row;
}

.ribbon {
  width: 10px;
  position: relative;
  display: inline-flex;
  flex-direction: column;
  border-top: 1px solid White;
  border-bottom: 1px solid White;
}

.arrow {
  height: 28px;
  position: absolute;
  z-index: 200;
	width: 0;
	height: 0;
}

.arrowTopBorder  {
	composes: arrow;
  border-right: none;
  border-top: none;
  border-bottom: 16px solid transparent;
  border-left: 10px solid White;
}

.arrowTop {
  composes: arrow;
  border-right: none;
  border-top: none;
  border-bottom: 14px solid transparent;
  border-left: 9px solid Pomegranate;
}

.arrowBottom {
	composes: arrow;
  top: 12px;
  border-right: none;
  border-top: 14px solid transparent;
  border-bottom: none;
  border-left: 9px solid Pomegranate;
}

.arrowBottomBorder {
	composes: arrowBottom;
  right: -1px;
  border-top: 15px solid transparent;
  border-bottom: none;
  border-left: 10px solid White;
}

.ribbonText {
  z-index: 10;
  display: inline-flex;
  align-items: center;
  height: 100%;
  justify-content: center;
  background-color: Pomegranate;
  border: 1px solid White;
  border-right: none;
  color: White;
  padding: 5px 11px 5px 15px;
  font-size: 12px;
  font-weight: 700;
  font-family: Avenir-Black, Avenir;
}
