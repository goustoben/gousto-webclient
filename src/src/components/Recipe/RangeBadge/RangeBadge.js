import React from 'react'
import Immutable from 'immutable'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import css from './RangeBadge.css'

const RangeBadge = ({ range, selectFoodBrand, isFoodBrandClickable }) => {
  let textColor
  let backgroundColor
  let borderColor
  let rangeSelect
  let rangeBadge
  let handleClick = null

  if (range && range.size) {
    rangeBadge = range.get('properties')

    textColor = rangeBadge.get('textColor')
    backgroundColor = rangeBadge.get('ribbonColor')
    borderColor = rangeBadge.get('borderColor')
    rangeSelect = {
      name: range.get('name'),
      slug: range.get('slug'),
      borderColor: backgroundColor
    }
  }

  const ribbonTextStyle = {
    color: textColor,
    backgroundColor: backgroundColor,
    border: `1px solid ${borderColor}`,
    borderRight: 'none',
    cursor: isFoodBrandClickable ? 'pointer' : 'default'
  }

  const ribbonStyle = {
    borderTop: `1px solid ${borderColor}`,
    borderBottom: `1px solid ${borderColor}`,
  }

  const arrowStyle = {
    borderLeft: `9px solid ${backgroundColor}`,
  }

  const arrowBorderStyle = {
    borderLeft: `10px solid ${borderColor}`,
  }
  if (isFoodBrandClickable) {
    handleClick = () => selectFoodBrand(rangeSelect)
  }

  return (rangeBadge) ? (
    <div className={css.rangeBadge}>
      <div
        role="button"
        tabIndex={isFoodBrandClickable ? 0 : -1}
        className={classnames(css.ribbonText, {[css.ribbonTextHighlight]: isFoodBrandClickable})}
        style={ribbonTextStyle}
        onClick={handleClick}
        onKeyPress={handleClick}
      >
        <span className={classnames(css.foodBrandName, {[css.foodBrandNameUnderlined]: isFoodBrandClickable})}>
          {range.get('name')}
        </span>
        {isFoodBrandClickable ? <span className={css.rightChevron}></span> : null}
      </div>
      <div className={css.ribbon} style={ribbonStyle}>
        <div className={css.arrowTopBorder} style={arrowBorderStyle}></div>
        <div className={css.arrowTop} style={arrowStyle}></div>
        <div className={css.arrowBottomBorder} style={arrowBorderStyle}></div>
        <div className={css.arrowBottom} style={arrowStyle}></div>
      </div>
    </div>
  ) : null
}

RangeBadge.propTypes = {
  range: PropTypes.instanceOf(Immutable.Map),
  selectFoodBrand: PropTypes.func,
  isFoodBrandClickable: PropTypes.bool,
}

export default RangeBadge
