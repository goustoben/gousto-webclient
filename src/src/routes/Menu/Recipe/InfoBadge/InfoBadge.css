@value Haricot, Radish, Carrot, White, DarkChilli, White, BlueCabbage, HaricotMedium from 'styles/colors.css';

.badge {
	composes: paddingSMLeftRight marginXSRight marginXSBottom from 'sass-loader!styles/spacing.scss';
	composes: inlineBlock from 'styles/display.css';
	composes: textXS from 'styles/typography.css';
	font-weight: bold;
	border-radius: 4px;
	border: 1px solid;
	padding-top: 0.125rem;
	padding-bottom: 0.125rem;
}

.newRecipe {
	color: Haricot;
	border-color: HaricotMedium;
}

.stockInfo {
	color: Radish;
	border-color: Radish;
}

.recommended {
	color: Carrot;
	border-color: Carrot;
}

.inverse {
	color: White;
	border: none;
	background-color: DarkChilli;
}

.chefPrepared {
	background-color: BlueCabbage;
	font-weight: normal;
	border: none;
}
