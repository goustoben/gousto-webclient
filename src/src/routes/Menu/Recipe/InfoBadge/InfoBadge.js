import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'
import css from './InfoBadge.css'

const InfoBadge = ({ type = 'span', children, newRecipe, stockInfo, recommended, inverse, chefPrepared, ...props }) => {
  const className = classNames(
    css.badge,
    { [css.newRecipe]: newRecipe },
    { [css.recommended]: recommended },
    { [css.inverse]: inverse },
    { [css.chefPrepared]: chefPrepared },
  )

  return React.createElement(type, { className, ...props }, children)
}

InfoBadge.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node,
  newRecipe: PropTypes.bool,
  stockInfo: PropTypes.bool,
  recommended: PropTypes.bool,
  inverse: PropTypes.bool,
  chefPrepared: PropTypes.bool
}

InfoBadge.defaultProps = {
  newRecipe: false,
  stockInfo: false,
  recommended: false,
  inverse: false,
  chefPrepared: false
}

export { InfoBadge }
