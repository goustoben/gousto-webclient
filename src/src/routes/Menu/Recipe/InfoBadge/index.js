import { InfoBadgeContainer } from './InfoBadgeContainer'
import { InfoBadgeSlugs } from './InfoBadgeSlugs'

export {
  InfoBadgeContainer,
  InfoBadgeSlugs
}
