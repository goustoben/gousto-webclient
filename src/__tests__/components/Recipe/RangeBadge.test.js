import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import sinon from 'sinon'

import RangeBadge from 'Recipe/RangeBadge'
import InfoBadge from 'Recipe/InfoBadge'
import colors from 'styles/colors.css'

describe('<RangeBadge />', () => {
  test('should not render by default', () => {
    const wrapper = shallow(<RangeBadge />)

    expect(wrapper.find(InfoBadge).length).toBe(0)
  })

  describe('snapshots', () => {
    test('should render when given a preconfigured range', () => {
      const whiteStub = sinon.stub(colors, 'White').get(() => '#FFFFFF')
      const tree = renderer
        .create(<RangeBadge range={'ten_to_table'} />)
        .toJSON()

      expect(tree).toMatchSnapshot()
      whiteStub.reset()
    })
  })

  describe('render', () => {
    test('should render badge for Ten to table', () => {
      const wrapper = shallow(<RangeBadge range={'ten_to_table'}/>)
      expect(wrapper.find('.rangeBadge').length).toEqual(1)
    })
    test('should render badge for Everydays favorites', () => {
      const wrapper = shallow(<RangeBadge range={'everyday_favourites'}/>)
      expect(wrapper.find('.rangeBadge').length).toEqual(1)
    })
  })
})
