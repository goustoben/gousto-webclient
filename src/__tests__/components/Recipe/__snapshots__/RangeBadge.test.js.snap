// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeBadge /> snapshots should render when given a preconfigured range 1`] = `
<div
  className="rangeBadge"
>
  <div
    className="ribbonText"
    style={
      Object {
        "backgroundColor": "#333D47",
        "border": "1px solid #282B2F",
        "borderRight": "none",
        "color": "#FFFFFF",
      }
    }
  >
    TEN TO TABLE
  </div>
  <div
    className="ribbon"
    style={
      Object {
        "borderBottom": "1px solid #282B2F",
        "borderTop": "1px solid #282B2F",
      }
    }
  >
    <div
      className="arrowTopBorder"
      style={
        Object {
          "borderLeft": "10px solid #282B2F",
        }
      }
    />
    <div
      className="arrowTop"
      style={
        Object {
          "borderLeft": "9px solid #333D47",
        }
      }
    />
    <div
      className="arrowBottomBorder"
      style={
        Object {
          "borderLeft": "10px solid #282B2F",
        }
      }
    />
    <div
      className="arrowBottom"
      style={
        Object {
          "borderLeft": "9px solid #333D47",
        }
      }
    />
  </div>
</div>
`;
